build.gradle.kts

```diff
@@ -355,6 +355,9 @@ tasks.register("generateBuildInfoClassFile") {
   val projectInfoDir = "${buildDir}/generated/logisim/java/com/cburch/logisim/generated"
   val projectInfoFile = "${projectInfoDir}/BuildInfo.java"
 
+  inputs.files(fileTree("src/main")).withPropertyName("sourceFiles")
```

Comment:
This doesn't seem to force a rebuild if you change the version in gradle.properties. I suggest something like: 

  inputs.dir("$projectDir/src")
  inputs.dir("$projectDir/support")
  inputs.dir("$projectDir/artwork")
  inputs.files("$projectDir/gradle.properties", "$projectDir/README.md", "$projectDir/LICENSE.md")

That seems to force a new timestamp when something going into the distribution packages changes. Of course, I still might have omitted something.

I note that this also hard-codes things. I don't currently have a solution that doesn't.

Reply:
BTW: I wonder if we should not fix the dependency for `compileJava` to contain:
```
inputs.files("$projectDir/gradle.properties", "$projectDir/README.md", "$projectDir/LICENSE.md")
```
because in fact it means we need to rebuild the app, not to generate a class (which is a consequence of building, not a reason for it).

Reply:
We would still need the dependsOn("genBuildInfo") since it must run before compileJava. I tried removing that and genBuildInfo doesn't even run when if I change gradle.properties and invoke a higher task such as createApp. Adding the inputs.files you suggest doesn't change that. Apparently, if genBuildInfo does run, and compileJava is requested, it actually recompiles because a source (the buildinfo) has changed. So I don't think it would hurt to add the line you suggest (as long as we leave the dependsOn), but I don't think it is technically necessary.